service/controller/app/v1/resource/chartoperator/resource.go

```diff
@@ -98,36 +95,45 @@ func (r Resource) installChartOperator(ctx context.Context, cr v1alpha1.App) err
 		return microerror.Mask(err)
 	}
 
-	// check app CR for chart-operator and fetching app-catalog name and version.
-	var tarballURL string
+	var chartOperatorAppCR *v1alpha1.App
 	{
-		r.logger.LogCtx(ctx, "level", "debug", "message", "finding a chart-operator app CR")
-		chartOperator, err := r.g8sClient.ApplicationV1alpha1().Apps(cr.Namespace).Get(release, metav1.GetOptions{})
-		if apierrors.IsNotFound(err) {
-			r.logger.LogCtx(ctx, "level", "debug", "message", "can't find a chart-operator app CR")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "finding chart-operator app CR")
 
+		chartOperatorAppCR, err = r.g8sClient.ApplicationV1alpha1().Apps(cr.Namespace).Get(release, metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "can't find chart-operator app CR")
 			r.logger.LogCtx(ctx, "level", "debug", "message", "cancelling the resource")
+			resourcecanceledcontext.SetCanceled(ctx)
 			return nil
 		} else if err != nil {
 			return microerror.Mask(err)
 		}
-		r.logger.LogCtx(ctx, "level", "debug", "message", "foung a chart-operator app CR")
 
-		catalogName := key.CatalogName(*chartOperator)
+		r.logger.LogCtx(ctx, "level", "debug", "message", "found chart-operator app CR")
+	}
 
-		r.logger.LogCtx(ctx, "level", "debug", "message", "finding a appCatalog CR")
-		chartCatalog, err := r.g8sClient.ApplicationV1alpha1().AppCatalogs().Get(catalogName, metav1.GetOptions{})
-		if apierrors.IsNotFound(err) {
-			r.logger.LogCtx(ctx, "level", "debug", "message", "can't find a appCatalog CR")
+	var appCatalogCR *v1alpha1.AppCatalog
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", "finding appCatalog CR")
 
+		catalogName := key.CatalogName(*chartOperatorAppCR)
+		appCatalogCR, err = r.g8sClient.ApplicationV1alpha1().AppCatalogs().Get(catalogName, metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "can't find appCatalog CR")
 			r.logger.LogCtx(ctx, "level", "debug", "message", "cancelling the resource")
+			resourcecanceledcontext.SetCanceled(ctx)
```

Comment:
if no chart-operator in TC, does it make sense to continue to other resources? Why not simply cancel the whole reconciliation?

Reply:
Good point thanks. I want to merge this but I'll do a follow up PR.